tree/treeplayer/src/TDFInterface.cxx

```diff
@@ -64,29 +65,43 @@ std::vector<std::string> FindUsedColumnNames(std::string_view expression, TObjAr
    }
 
    // Check which data-source columns match
-   for (auto col : dsColumns) {
+   for (auto &col : dsColumns) {
       std::string bNameRegexContent = regexBit + col + regexBit;
       TRegexp bNameRegex(bNameRegexContent.c_str());
       if (-1 != bNameRegex.Index(paddedExpr.c_str(), &paddedExprLen)) {
-         // if not already found among the custom columns
+         // if not already found among the other columns
          if (std::find(usedBranches.begin(), usedBranches.end(), col) == usedBranches.end())
```

Comment:
What happens when it's already defined? Should we redefine? It looks like this silently ignores already defined columns, which may be unintuitive for users.

Reply:
The redefinition or, more in general, bogus definition is handled by TLoopManager::AddColumnAlias: https://github.com/root-project/root/pull/1048/files#diff-600cca72f8a33c06aa6afa8f9fdfc4e4R481 
The case of redefinition of an alias to a different column is considered